### PR TITLE
tweaked the create_db_and_tables to enable table exclusions

### PR DESCRIPTION
**Pull request by yoann-diep**
**Type of change**
- [*] Feature
- [ ] Bugfix
- [ ] Refactoring
- [ ] Documentation

-----------------------

**Context**

>_What does this PR implement and how? What are the requirements?_

>_Why is this change needed? Related issue #?_

>_Any diagrams or screenshot to help reviewers?_

>_Please describe the tests you have performed to ensure the feature or fix are robust & effective?_

Your comments

-----------------------

**What should the reviewer focus on?**
>_Do you have specific questions/ areas for the reviewer?_

>_Which alternative solutions have you already considered and why did you not implement it?_

>_Where is the highest risk/ most complicated change that the reviewer should focus on?_

>_Is there a specific code review topic the reviewer should focus on (e.g., correct error handling, API usage)_

>_What is a good place to start the review (e.g., specific file/method?)_

Your comments

-----------------------

**For reviewer:**

[PR Review guidelines](https://google.github.io/eng-practices/review/reviewer/standard.html)

### CODE REVIEW

File: ecodev_core/db_connection.py [hunk 1]
```diff
@@ -34,11 +36,15 @@ class DbSettings(BaseSettings):
 engine = create_engine(DB_URL)
 
 
-def create_db_and_tables(model: Callable) -> None:
+def create_db_and_tables(model: Callable, excluded_tables: Optional[List[str]] = None) -> None:
     """
     Create all tables based on the declared schemas in core/models thanks to sqlmodel
```
Comment (tepelbaum) [inline ecodev_core/db_connection.py:41]: Update the docstring accodingly :)

File: ecodev_core/db_connection.py [hunk 2]
```diff
@@ -34,11 +36,15 @@ class DbSettings(BaseSettings):
...
     """
     log.info(f'Inserting on the fly {model} and all other domain tables')
+    if excluded_tables:
+        included_tables = [table for table in SQLModel.metadata.tables if
+                           table not in excluded_tables]
+        SQLModel.metadata.tables = {k: SQLModel.metadata.tables[k] for k in included_tables}
```
Comment (tepelbaum) [inline ecodev_core/db_connection.py:47]: Do you have a source for this?

Comment (yoann-diep) [reply ecodev_core/db_connection.py:47]: what do you mean by source?

Comment (tepelbaum) [inline ecodev_core/db_connection.py:47]:
```suggestion
    if excluded_tables:
        included_tables = [table for table in SQLModel.metadata.tables if
                           table not in excluded_tables]
        SQLModel.metadata.tables = {table: meta_data for for table, meta_data in SQLModel.metadata.table.items() if table not in excluded_tables}
```

Comment (tepelbaum) [reply ecodev_core/db_connection.py:47]: rebase on main as well :)

**Pull request merged**